File: Game-In-Java/src/GameGUI.java
import javax.swing.*;
import java.awt.*;
import java.awt.event.ActionEvent;
import java.awt.event.ActionListener;

public class GameGUI {

    private JFrame frame;
    private JTextField guessField;
    private JLabel resultLabel;
    private Game game;

    public GameGUI(){
        game = new Game();
        createAndShowGUI();
    }

    private void createAndShowGUI(){
        // Utworzenie okna głównego
        frame = new JFrame("Zgadnij liczbę");
        frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
        frame.setLayout(new BorderLayout());

        // Utworzenie panelu głównego
        JPanel panel = new JPanel();
        panel.setLayout(new GridLayout(3, 1));

        // Pole tekstowe do wpisania liczby
        guessField = new JTextField();
        panel.add(new JLabel("Podaj swoją liczbę (1-100)"));
        panel.add(guessField);

        // Przycisk zgadywania
        JButton guessButton = new JButton("Zgadnij!");
        guessButton.addActionListener(new GuessButtonListener());
        panel.add(guessButton);

        // Etykieta wyniku
        resultLabel = new JLabel("Wynik bedzie tutaj.", SwingConstants.CENTER);
        frame.add(resultLabel, BorderLayout.SOUTH);

        frame.add(panel, BorderLayout.CENTER);

        // Wyświetlanie okna
        frame.setVisible(true);
    }

    private class GuessButtonListener implements ActionListener {
        @Override
        public void actionPerformed(ActionEvent e) {
            try{
                int guess = Integer.parseInt(guessField.getText());
                String result = game.checkGuess(guess);
                resultLabel.setText(result);

                // Sprawdzenie czy gra się zakończyła
                if(result.startsWith("Gratulacje!")){
                    int option = JOptionPane.showConfirmDialog(frame, "Czy chcesz zagrać ponownie?", "Koniec gry", JOptionPane.YES_NO_OPTION);
                    if(option == JOptionPane.YES_OPTION){
                        game.resetGame();
                        resultLabel.setText("Wynik będzie tutaj.");
                        guessField.setText("");
                    } else{
                        System.exit(0);
                    }
                }
            } catch (NumberFormatException ex){
                resultLabel.setText("Proszę wprowadzić poprawną liczbę");
            }
        }
    }
}

File: Game-In-Kotlin/src/main/kotlin/GameGUI.kt
import java.awt.BorderLayout
import java.awt.GridLayout
import javax.swing.*

class GameGUI {
    private val frame: JFrame = JFrame("Zgadnij liczbę")
    private val guessField: JTextField = JTextField()
    private val resultLabel: JLabel = JLabel("Wynik będzie tutaj.", SwingConstants.CENTER)
    private val game: Game = Game()

    init{
        createAndShowGUI()
    }

    private fun createAndShowGUI(){
        frame.defaultCloseOperation = JFrame.EXIT_ON_CLOSE
        frame.setSize(400, 200)
        frame.layout = BorderLayout()

        val panel = JPanel()
        panel.layout = GridLayout(3, 1)

        panel.add(JLabel("Podaj swoją liczbę (1-100):"))
        panel.add(guessField)

        val guessButton = JButton("Zgadnij")
        guessButton.addActionListener{
            try{
                val guess = guessField.text.toInt()
                val result = game.checkGuess(guess)
                resultLabel.text = result

                if(result.startsWith("Gratulacje!")){
                    val option = JOptionPane.showConfirmDialog(frame, "Czy chcesz zagrać ponownie?", "Koniec Gry", JOptionPane.YES_OPTION)
                    if(option == JOptionPane.YES_OPTION){
                        game.resetGame()
                        resultLabel.text = "Wynik będzie tutaj."
                        guessField.text = ""
                    } else{
                        System.exit(0)
                    }
                }
            } catch (ex: NumberFormatException){
                resultLabel.text = "Proszę wprowadzić poprawną liczbę"
            }
        }

        panel.add(guessButton)

        frame.add(panel, BorderLayout.CENTER)
        frame.add(resultLabel, BorderLayout.SOUTH)

        frame.isVisible = true
    }
}